### PR TITLE
Adding license information for Openbookcorpus

### PR DESCRIPTION
**Pull request by meg-huggingface**
Not entirely sure, following the links here, but it seems the relevant license is at https://github.com/soskek/bookcorpus/blob/master/LICENSE

### CODE REVIEW

**Comment by severo** [comment]
Thanks, @meg-huggingface for the updates!

Thanks also for setting me as a reviewer, but I'm totally not a specialist of the datasets themselves, so I prefer to just lurk and let @lhoestq @albertvillanova @mariosasko or @patrickvonplaten review these changes (https://github.com/huggingface/datasets/pulls/assigned/meg-huggingface).

**Comment by lhoestq** [comment]
The MIT license seems to be for the crawling code, no ? Then maybe we can also redirect users to the [terms of smashwords.com](https://www.smashwords.com/about/tos) regarding copyrights, in particular the paragraph 10 for end-users. In particular it seems that end users can download and use the content "for their personal enjoyment in any reasonable non-commercial manner in compliance with copyright law" and the smashwords end-users agreement.

**Review by severo — COMMENTED** [review]
Thanks, @meg-huggingface for the updates!

Thanks also for setting me as a reviewer, but I'm totally not a specialist of the datasets themselves, so I prefer to just lurk and let @lhoestq @albertvillanova @mariosasko or @patrickvonplaten review these changes.

File: datasets/bookcorpusopen/README.md [hunk 1]
```diff
@@ -158,7 +158,7 @@ The data fields are the same among all splits.
 
 ### Licensing Information
 
-[More Information Needed](https://github.com/huggingface/datasets/blob/master/CONTRIBUTING.md#how-to-contribute-to-the-dataset-cards)
+[MIT License](https://github.com/soskek/bookcorpus/blob/master/LICENSE)
```
Comment (lhoestq) [inline datasets/bookcorpusopen/README.md:161]: Sorry for the delay, let's at least add some relevant redirections here:

```suggestion
The books have been crawled from smashwords.com, see their [terms of service](https://www.smashwords.com/about/tos) for more information.

A data sheet for this dataset has also been created and published in [Addressing "Documentation Debt" in Machine Learning Research: A Retrospective Datasheet for BookCorpus](https://arxiv.org/abs/2105.05241)
```

**Comment by lhoestq** [comment]
May I merge this one ?

**Pull request merged**